script/enforce_type_hints.py

```diff
@@ -0,0 +1,270 @@
+#!/usr/bin/env python3
```

Comment:
Lets solve this using existing linters instead of doing this DIY scripting style.